4.applications/grit-informed-aggregation/1.generate-profiles/scripts/eval_utils.py

```diff
@@ -0,0 +1,189 @@
+import os
+import glob
+import pandas as pd
+import numpy as np
+import random
+
+from pycytominer.cyto_utils import infer_cp_features
+
+
+def get_featuredata(df):
+    """Return dataframe of output features that we expect to be cell painting features"""
+    return df[infer_cp_features(df, metadata=False)]
+
+
+def get_metadata(df):
+    """Return dataframe of just metadata columns"""
+    return df[infer_cp_features(df, metadata=True)]
+
+
+def percent_score(null_dist, corr_dist, how):
+    """
+    Calculates the Percent strong or percent recall scores
+
+    Arguments:
+    null_dist - Null distribution
+    corr_dist - Correlation distribution
+    how - ["left", "right" or "both"] for using the 5th percentile, 95th percentile or both thresholds
+
+    Return:
+    Proportion of correlation distribution beyond the threshold
+    """
+    if how == "right":
+        perc_95 = np.nanpercentile(null_dist, 95)
+        above_threshold = corr_dist > perc_95
+        return np.mean(above_threshold.astype(float)), perc_95
+    if how == "left":
+        perc_5 = np.nanpercentile(null_dist, 5)
+        below_threshold = corr_dist < perc_5
+        return np.mean(below_threshold.astype(float)), perc_5
+    if how == "both":
+        perc_95 = np.nanpercentile(null_dist, 95)
+        above_threshold = corr_dist > perc_95
+        perc_5 = np.nanpercentile(null_dist, 5)
+        below_threshold = corr_dist < perc_5
+        return (
+            np.mean(above_threshold.astype(float))
+            + np.mean(below_threshold.astype(float)),
+            perc_95,
+            perc_5,
+        )
+
+
+def corr_between_replicates(df, group_by_feature):
```

Comment:
add the same note to any script that you modified from Niranj

Reply:
added in 335d81483e54227b0b9d61373582b124ee1dba2e